src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -85,18 +85,27 @@ private SingleNamespaceDeclaration GetMatchingNamespaceDeclaration(CSharpSyntaxN
             throw ExceptionUtilities.Unreachable();
         }
 
-        private static AliasesAndUsings GetOrCreateAliasAndUsings(
-            ref ImmutableSegmentedDictionary<SingleNamespaceDeclaration, AliasesAndUsings> dictionary,
+        private AliasesAndUsings GetOrCreateAliasAndUsings(
+            SegmentedDictionary<SingleNamespaceDeclaration, AliasesAndUsings> dictionary,
             SingleNamespaceDeclaration declaration)
         {
-            return RoslynImmutableInterlocked.GetOrAdd(
-                ref dictionary,
-                declaration,
-                static _ => new AliasesAndUsings());
+            if (!dictionary.TryGetValue(declaration, out AliasesAndUsings? result))
```

Comment:
this is not threadsafe.  you're reading a mutable structure outside the lock while something might be in the lock mutating it.